### ISSUE TITLE
Gum should make it easier to make complex outer components

### ISSUE THREAD

**Issue by profexorgeek** [vchelaru/FlatRedBall#1067]
There are two use cases that have the same underlying problem and I believe they are a barrier to making even more awesome Gum UI.

**The TLDR is: It should be easy to create "wrapping components" that are designed to hold other components. For example, list boxes, menu panels, and other parent containers.**

The first use case is putting components into a ListBox at design time (as opposed to dynamically with MVVM). I recently created complex menus for Masteroid and to prevent overflow on small-resolution devices, I put parts of the UI into ListBoxes. The workflow for this is clunky and problematic. Here are the steps the user must take:

1. Add a ListBox to the component
2. Add the components you want in the ListBox
3. Set the component parent to the ListBoxInstance.InnerPanelInstance
  - The first big problem occurs here. As you try to type in this box, the UI behaves erratically and often when you press the period key the input box loses focus and you have to click into the box again and re-enter your text.
4. Observe the child entities in the list view: their list view representation does not reflect how they are actually parented

The second use case is similar. Let's say I want a menu background that makes it easy to make menus with a consistent look and feel. This menu background component might have: title, close button, background nine slice, border nine slice, and a list container designed to hold buttons or menu items. I should be able to drag an instance of the menu background into various menu components and easily add buttons to the inner list container - similar to how a ListBox works. Instead, I have to do the same manual parenting dance and end up with a treeview that is fragile and doesn't reflect the actual parent/child relationship of the components.

**Recommended implementation**: Gum components already allow you to expose variables to customize the interface available on instances of the component. Gum components should have a new special property such as `IsPrimaryContainer` which allows the designer to specify a single container within the component as the default place where children are stored. This would allow designers to create really intricate components with fancy borders and layout, but other team members could easily add children to instances of that component without understanding its inner layout. This would also make ListBoxes, ComboBoxes and similar controls easier to use when adding design-time children!

**Comment by profexorgeek** [comment]
Another use case is something like an item frame. It might have borders, inner shadows, etc. And a special clipping container that clips overflowing icons (I do this in both Masteroid and Wee to show item icons without worrying about weirdly-shaped weapons overflowing).

You would be able to drag/drop an image into the special item frame and it would land in the right clipping container and display correctly in the treeview!

**Comment by vchelaru** [comment]
To make it clear that only one child can be the primary, we'll have this be a property on the Component itself.

The property will be `DefaultChildrenContainer`

**Comment by vchelaru** [comment]
This has been tested and seems to be working well.

Please note that this feature is (currently) only supported in Gum and not FRB codegen.
